src/main/java/seedu/storemando/logic/parser/ReminderCommandParser.java

```diff
@@ -0,0 +1,61 @@
+package seedu.storemando.logic.parser;
+
+import static seedu.storemando.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.storemando.logic.commands.ReminderCommand;
+import seedu.storemando.logic.parser.exceptions.ParseException;
+import seedu.storemando.model.item.ItemExpiringPredicate;
+
+/**
+ * Parses input arguments and creates a new ReminderCommand object
+ */
+public class ReminderCommandParser implements Parser<ReminderCommand> {
+    /**
+     * Convert the given number and the time unit to the number of days.
+     * @param num The number use to covert to days
+     * @param timeUnit The time unit in terms of days and weeks
+     * @return The number of days
+     * @throws ParseException if the user input does not conform the expected keyword
+     */
+    private long timeConversation(long num, String timeUnit) throws ParseException {
+        switch (timeUnit) {
+        case "days":
+            return num;
+        case "weeks":
+            return num * 7;
+        default:
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ReminderCommand.MESSAGE_USAGE));
+        }
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ReminderCommand
+     * and returns a ReminderCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    @Override
+    public ReminderCommand parse(String args) throws ParseException {
+        try {
+            long numOfDayFromToday = 0;
+            String trimmedArgs = args.trim();
+            String[] stringArgsArr = trimmedArgs.split(" ");
```

Comment:
I think this way of checking would break if the user keys in "reminder  3" with 2 spaces instead of 1. This way " 3" your argument would be " 3" which would cause a number format exception which would be caught and eventually be thrown as a parse exception. Are we going to be strict about this? Currently, the other commands don't seem to take into account additional spaces in between prefixes and command word.

Reply:
Can you clarify the way you are parsing?

Reply:
This wouldn't be the case. It will still works regardless how many spaces are in between the reminder and the number.
Eg. "Reminder                 3" would work the same as "Reminder 3"

Reply:
Basically, I abstract the user input string into arrays of string. The stringArgsArr[0] has to be a number in a string format and the stringArgsArr[1] has to be a "days" or "weeks". With these 2 valid arguments, I will proceed to parse the number of days in total. If no days or weeks are declared in the user input, by default the number given is in timeUnit days.